innopoints/views/application.py

```diff
@@ -95,6 +102,51 @@ def take_back_application(project_id, activity_id):
     return NO_PAYLOAD
 
 
+@api.route('/projects/<int:project_id>/activities/<int:activity_id>'
+           '/applications/<int:application_id>', methods=['PATCH'])
+@login_required
+def change_application_status(project_id, activity_id, application_id):
+    """Change the status of an application."""
+    if not request.is_json:
+        abort(400, {'message': 'The request should be in JSON.'})
+
+    application = Application.query.get_or_404(application_id)
+    activity = Activity.query.get_or_404(activity_id)
+    project = Project.query.get_or_404(project_id)
+
+    if activity.project != project or application.activity_id != activity.id:
+        abort(400, {'message': 'The specified project, activity and application are unrelated.'})
+
+    if current_user not in project.moderators and not current_user.is_admin:
+        abort(401)
+
+    if project.lifetime_stage != LifetimeStage.ongoing:
+        abort(400, {'message': 'The application status may only be changed for ongoing projects.'})
+
+    try:
+        status = getattr(ApplicationStatus, request.json['status'])
+    except (KeyError, AttributeError):
+        abort(400, {'message': 'A valid application status must be specified.'})
+
+    if application.status != status:
+        application.status = status
+
+        notify(application.applicant_email, NotificationType.application_status_changed, {
+            'activity_id': activity_id,
+            'application_id': application_id,
+        })
```

Comment:
move the notification below the try/except block so that it doesn't send if the commit fails